zipline/utils/events.py

```diff
@@ -482,51 +436,27 @@ class NDaysBeforeLastTradingDayOfWeek(TradingDayOfWeekRule):
     def __init__(self, n):
         super(NDaysBeforeLastTradingDayOfWeek, self).__init__(n, invert=True)
 
-    @staticmethod
-    def get_last_trading_day_of_week(dt, cal):
-        prev = None
-        # Traverse forward until we hit a week border, then jump back to the
-        # previous trading day.
-        try:
-            while not prev or dt.weekday() > prev.weekday():
-                prev = dt
-                dt = cal.next_trading_day(dt)
-        except NoFurtherDataError:
-            prev = dt
-
-        if cal.is_open_on_day(prev):
-            return prev
-        else:
-            return cal.previous_trading_day(prev)
-
-    date_func = get_last_trading_day_of_week
-
 
 class TradingDayOfMonthRule(six.with_metaclass(ABCMeta, StatelessRule)):
     def __init__(self, n, invert):
         if not 0 <= n < MAX_MONTH_RANGE:
             raise _out_of_range_error(MAX_MONTH_RANGE)
-        self.month = None
-        self.date = None
-        self.td_delta = -n if invert else n
+        if invert:
+            self.td_delta = -n - 1
```

Comment:
Because the rule that uses invert is `NDaysBeforeLastTradingDayOfMonth`.  if you pass in 0 for that rule, you want the last business day of each month.  if you pass in -1 for that rule, you want the day before the last business day of each month.

but since internally we're using pandas `nth`, passing in 0 gives you the first day of the month, and -1 the last day of the month.  we subtract one to get the right day.